libs/hmpb/src/all_bubble_ballot_fixtures.tsx

```diff
@@ -332,6 +338,13 @@ export const allBubbleBallotFixtures = (() => {
               blankBallot,
               votesForSheet
             );
+            await injectCustomFooterMetadata(
+              sheetDocument,
+              <FooterMetadataContent
+                label="Cycle Index"
+                value={sheetIndex + 1}
```

Comment:
Unless I'm misreading something, cycle index should be computable from `pageNumber`, which the footer has as a prop. Something like: `Math.floor(pageNumber / 2) + 1`. That means it doesn't need to be injected after the initial render, but rather could be computed in the footer component.

Reply:
Ah yes, I should've added a note about this as it's not intuitive. But the all-bubble ballot is still just a 1-sheet (2-page) ballot. The cycling test deck just includes 6 sheets of this 1-sheet ballot. So the page number will always be 1 or 2.

| Cycling test deck sheet | Page number |
| - | - |
| 1 | 1 |
| | 2|
| 2 | 1 |
| | 2|
| 3 | 1 |
| | 2|
| 4 | 1 |
| | 2|
| 5 | 1 |
| | 2|
| 6 | 1 |
| | 2|

So the "Cycle Index" has to be injected at the test-deck-generation level as opposed to the individual ballot-generation level.

Reply:
One more reason for the injection was to only add the cycle index for the cycling test deck but not the blank all-bubble ballot. Though there's an argument to be made that we should always have the cycle index field there and just set it to, say, N/A when rendering the blank or filled all-bubble ballot.

Reply:
Ah right, ok I just read the code more fully and refreshed myself. Thanks for the explanation.

I remember previously just using the first marked bubble in the top row as my indicator of cycle index. But a printed number is a bit easier to use.

I think injecting it is fine given the requirements.

Reply:
> I remember previously just using the first marked bubble in the top row as my indicator of cycle index.

Yeah I've done the same. I don't have a great read on how exactly SLI manages and scans their all-bubble ballots for the accuracy test, but the general sense I got from Matt is that this would be helpful and make testing easier for SLI